pandas/io/packers.py

```diff
@@ -255,17 +261,36 @@ def unconvert(values, dtype, compress=None):
     if not as_is_ext:
         values = values.encode('latin1')
 
-    if compress == u'zlib':
-        import zlib
-        values = zlib.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
-
-    elif compress == u'blosc':
-        import blosc
-        values = blosc.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
+    if compress:
+        if compress == u'zlib':
+            from zlib import decompress
+        elif compress == u'blosc':
+            from blosc import decompress
+        else:
+            raise ValueError("compress must be one of 'zlib' or 'blosc'")
+
+        values = decompress(values)
+        if sys.getrefcount(values) == 2:
```

Comment:
The implementation proposed in this PR is scary so I tried to solve this problem upstream in  [blosc](https://github.com/Blosc/python-blosc/pull/107) and [zlib](http://bugs.python.org/issue26379)

Hopefully we can swap to the more supported solutions when available and only use this hack when we need to.